services/cd-service/pkg/repository/repository.go

```diff
@@ -1865,24 +1865,54 @@ func (s *State) DeleteDirIfEmpty(directoryName string) (SuccessReason, error) {
 	return DirNotEmpty, nil
 }
 
-func (s *State) GetQueuedVersion(environment string, application string) (*uint64, error) {
+func (s *State) GetQueuedVersionFromDB(ctx context.Context, transaction *sql.Tx, environment string, application string) (*uint64, error) {
+	queuedDeployment, err := s.DBHandler.DBSelectLatestDeploymentAttempt(ctx, transaction, environment, application)
+
+	if err != nil || queuedDeployment == nil {
+		return nil, err
+	}
+
+	var v uint64
+	if queuedDeployment.Version != nil {
+		v = uint64(*queuedDeployment.Version)
+	} else {
+		return nil, nil
```

Comment:
either replace this `return` with  ` v = nil` 
or
replace the `then` branch with `return ...`
for consistency